torchdynamo/variables/torch.py

```diff
@@ -148,6 +149,56 @@ def call_function(
         elif not config.dynamic_shapes and self.is_dynamic_shapes(args, kwargs):
             unimplemented(f"dynamic shapes: {self.value.__name__}")
         else:
+            # print('start of torch.py else branch')
+            # TODO: this is just a hack to get an e2e example working,
+            # needs a lot of cleanup before review
+            if self.value == torch.nn.functional.sigmoid:
+                # this extracts the tensor under test
+                # TODO: is this always there and can we depend on it?
+                tensor_val = args[0].proxy.node.meta['example_value']
+                from torch.overrides import has_torch_function_unary
+                has_tf = has_torch_function_unary(tensor_val)
+                if has_tf:
+                    from torchdynamo.source import LocalSource
+                    from torchdynamo.variables.builder import VariableBuilder, TupleVariable
+                    # note: does this actually need to be a UserFunctionVariable?
+                    # note: the __func__ attribute call is to ensure the function
+                    # does not have `self`
+                    func = tensor_val.__torch_function__.__func__
+
+                    # For now, we inline the object instance's __torch_function__ directly
+                    # In the future, it might be worth considering inlining
+                    # the `torch.overrides.handle_torch_function` method instead.
+
+                    # first, get the source
+                    # TODO: is this the right type of source to use?
+                    source = LocalSource('__torch_function__')
+                    # next, create the variable tracker
+                    func_var = VariableBuilder(tx, source)(func)
```

Comment:
The guard (reading the invalid value) is implicitly being generated by this call.  

I don't think we actually need a guard here.  [We already guard on the Python class of the tensor](https://github.com/facebookresearch/torchdynamo/blob/688448565e36f1e295fd213d6ab74b23e1faa4d9/torchdynamo/_guards.cpp#L143), so unless someone monkey patches `__torch_function__` it should be safe to instead do:
```suggestion
                    func_var = UserMethodVariable(func, <cls/self VariableTracker>, **options)
```
Which won't add any new guards (but propagate existing guards through `options`).